### PR TITLE
Update TiDB/TiKV to  4.0.0-rc config

### PR DESCRIPTION
**Pull request by Yisaer**
Update TiDB/TiKV to 4.0.0-rc config and I also found some configs missiong `omitempty` so I add it.

### What problem does this PR solve? 

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Update `TiDBConfig` and `TiKVConfig` to support `4.0.0-rc` version
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/tidb_config.go [hunk 1]
```diff
@@ -50,6 +50,10 @@ type TiDBConfig struct {
 	// Optional: Defaults to 1000
 	// +optional
 	TokenLimit *uint `toml:"token-limit,omitempty" json:"token-limit,omitempty"`
+	// +optional
+	OOMUseTmpStorage *bool `toml:"oom-use-tmp-storage" json:"oom-use-tmp-storage,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tidb_config.go:54]: No need to `omitempty` for toml?

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidb_config.go]: updated.

**Pull request merged**